### PR TITLE
Parse skills map

### PR DESCRIPTION
**Pull request by riknoll**
Very simple syntax:

Single # header defines the skills map
Double ## header defines an activity
Key value pairs are just `key: value` and are case insensitive for the keys

For map prerequisites, they can be either skills maps you must have finished or tags

### CODE REVIEW

File: skillsmap/src/lib/skillMap.d.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+interface ActivityPrerequisite {
+    type: "activity";
+    mapId: string;
+}
+
+interface MapActivity {
+    activityId: string;
+
+    displayName: string;
+    description: string;
```
Comment (shakao) [inline skillsmap/src/lib/skillMap.d.ts:28]: required?

File: skillsmap/src/lib/skillMap.d.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+interface MapState {
+    mapId: string;
+    activityState: {[index: string]: ActivityState};
+}
+
+interface ActivityState {
+    activityId: string;
+    headerId: string;
```
Comment (shakao) [inline skillsmap/src/lib/skillMap.d.ts:53]: just checking: this is so we can load them back into a partially completed activity?

File: skillsmap/src/lib/skillMapParser.ts [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+}
+
+export function test() {
+    return parseSkillsMap(testMap);
+}
+
+export function parseSkillsMap(text: string) {
+    const sections = getSectionsFromText(text);
+
+    const result = inflateSkillMap(sections[0]);
```
Comment (shakao) [inline skillsmap/src/lib/skillMapParser.ts:58]: later we would expand this to theoretically return an array of skill maps?

File: skillsmap/src/lib/skillMap.d.ts (hunk 2 above)
Comment (riknoll) [reply skillsmap/src/lib/skillMap.d.ts:53]: yep!

File: skillsmap/src/lib/skillMap.d.ts (hunk 1 above)
Comment (riknoll) [reply skillsmap/src/lib/skillMap.d.ts]: Changed

**Review by shakao — APPROVED** [review]

**Pull request merged**